src/targets/node/terminalNodeLauncher.ts

```diff
@@ -24,6 +24,10 @@ import { hideDebugInfoFromConsole, INodeBinaryProvider, NodeBinary } from './nod
 import { IProcessTelemetry, IRunData, NodeLauncherBase } from './nodeLauncherBase';
 import { IProgram } from './program';
 
+const SHELL_INTEGRATION_TIMEOUT = 1000;
+const SHELL_PROCESS_ID_TIMEOUT = 1000;
+const SHELL_INTEGRATION_SETTING = 'terminal.integrated.shellIntegration.enabled';
```

Comment:
nit: `const enum` is great for this type of thing, as the values are inlined and are then grouped and can have jsdoc in code.